run-tests.php

```diff
@@ -3611,7 +3611,13 @@ public function checkSkip(string $php, string $code, string $checkFile, string $
         }
 
         save_text($checkFile, $code, $tempFile);
-        $result = trim(system_with_timeout("$php \"$checkFile\"", $env));
+        $useSubprocess = $testHasIniSection || shouldRunInSubprocess($code);
```

Comment:
`$php` is not necessarily the same as the currently running PHP binary. This should be handled accordingly.